### PR TITLE
Bump 2FA secret bit length from 80 to 160 bits as recommended by RFC4226

### PR DESCRIPTION
**Pull request by evilaliv3**
Implement fix for ticket https://github.com/freedomofpress/securedrop/issues/5933

### CODE REVIEW

**Comment by eloquence** [comment]
Hi @evilaliv3, thanks for working on this! The lint failure is real due to missing type annotations - would you be comfortable adding those, or would you like us to do so?

File: securedrop/models.py [hunk 1]
```diff
@@ -403,7 +407,7 @@ class Journalist(db.Model):
     is_admin = Column(Boolean)  # type: Column[Optional[bool]]
     session_nonce = Column(Integer, nullable=False, default=0)
 
-    otp_secret = Column(String(16), default=pyotp.random_base32)
```
Comment (zenmonkeykstop) [inline securedrop/models.py:406]: latest version of pyotp (2.6.0) now defaults to a 32-char base32 string from random_base32(). This should definitely be updated but it might make sense to update the dependency and the relevant references in the codebase. Any thoughts on that approach @evilaliv3 ?

**Comment by zenmonkeykstop** [comment]
(moved comment to issue: https://github.com/freedomofpress/securedrop/issues/5933#issuecomment-844473763 )

File: securedrop/models.py (hunk 1 above)
Comment (evilaliv3) [reply securedrop/models.py:406]: Thank you for noticing this @zenmonkeykstop , you are right the size of this string should be changed accordingly.

I actually notice to have to do this on our codebase as well and as a fun fact SQLITE seems to ignore the size limit (probably due to default allogators on Strings that always allocate more than just 16 chars)

**Comment by evilaliv3** [comment]
@zenmonkeykstop  @eloquence : Either the two options are fine for me.

Among the two options I support dropping the pyotp dependency for the benefit of limiting application dependencies.

In any case I've updated the pull request to reflect your pull request review and advice! :)

File: securedrop/models.py (hunk 1 above)
Comment (zenmonkeykstop) [reply securedrop/models.py:406]: yup varchars are humongous, had some "fun" last night writing db migration tests for what is effectively a no-op.

File: install_files/ansible-base/roles/restore/tasks/perform_restore.yml [hunk 2]
```diff
@@ -23,8 +23,10 @@
   connection: local
   become: no
   unarchive:
+    # Avoid creating an unnecessary local copy
+    remote_src: yes
     dest: "{{ torrc_check_dir.path }}/backup/"
-    src: "{{ restore_file }}"
+    src: "{{playbook_dir}}/{{ restore_file }}"
```
Comment (zenmonkeykstop) [inline install_files/ansible-base/roles/restore/tasks/perform_restore.yml]: this is unrelated to the PR, I wonder if a rebase went awry somewhere. will take a look.

File: securedrop/models.py [hunk 3]
```diff
@@ -35,6 +35,10 @@
 ARGON2_PARAMS = dict(memory_cost=2**16, rounds=4, parallelism=2)
 
 
+def generate_otp_secret() -> bytes:
```
Comment (zenmonkeykstop) [inline securedrop/models.py]: Type checking is failing, as `bytes` are specified but a `str` is expected in the `models.py` call.

**Review by zenmonkeykstop — CHANGES_REQUESTED** [review]

File: securedrop/models.py (hunk 3 above)
Comment (evilaliv3) [reply securedrop/models.py]: It should be fixed now

File: install_files/ansible-base/roles/restore/tasks/perform_restore.yml (hunk 2 above)
Comment (evilaliv3) [reply install_files/ansible-base/roles/restore/tasks/perform_restore.yml]: Fixed preparing the pull request for integration on current develop branch

File: securedrop/models.py [hunk 4]
```diff
@@ -35,6 +35,10 @@
 ARGON2_PARAMS = dict(memory_cost=2**16, rounds=4, parallelism=2)
 
 
+def generate_otp_secret() -> str:
+    return base64.b32encode(os.urandom(20)).decode()
```
Comment (zenmonkeykstop) [inline securedrop/models.py:39]: Since the move to Focal SD can probably use secrets (Python 3.6+) now, so you could also do something like:

```
import string
import secrets

def generate_otp_secret() -> str:
    symbols = string.ascii_uppercase + string.digits
    return ''.join(secrets.choice(symbols) for i in range(OTP_LENGTH))
```

Might be good to abstract away from os.urandom, and it achieves the goal of taking pyotp out of the requirements.

Comment (evilaliv3) [reply securedrop/models.py:39]: I agree actually. would you like to author this ?

**Comment by evilaliv3** [comment]
Closing this pull in order to open a clean one with a different implementation.
